### ISSUE TITLE
new lint: "no fancy createState"

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#1875]
Follow-up from https://github.com/dart-lang/linter/pull/1867 (and related to: https://github.com/dart-lang/linter/pull/1865).

**Bad:**

```dart
MyState global;

class MyStateful extends StatefulWidget {
  @override
  MyState createState() {
    global = MyState();
    return global;
  } 
}
```

**Good:**

```dart
class MyStateful extends StatefulWidget {
  @override
  MyState createState() {
    return MyState();
  }
}
```

**Comment by pq** [comment]
Naming is hard.  @rrousselGit, @bwilkerson: any ideas?

`no_state_create_logic`  ?

**Comment by bwilkerson** [comment]
`no_logic_in_create_state`?

**Comment by pq** [comment]
@rrousselGit:

how literally should we interpret this advice?


> Ultimately, createState has just a single possible implementation, that is:

```dart
@override
MyState createState() {
  return MyState();
}
```

> With maybe replacing {} + return with () =>.
> But other than that, it shouldn't change, ever.

Literally should we only except a block w/ one statement (a return statement instantiating a new State instance)?

**Comment by rrousselGit** [comment]
Yes, with no constructor parameter, no if, ternary, cascade...

**Comment by bwilkerson** [comment]
That does appear to trump the previous lint.
